include/FilesystemManagement.h

```diff
@@ -0,0 +1,156 @@
+        NameTooLong,       ///< The name of a file or directory specified is too long.
+        // Permission Errors
+        PermissionDenied,  ///< The filesystem permissions forbid the action to be taken.
+        ReadOnly,          ///< The filesystem only permits read operations.
+        // Directory Errors
+        NotADirectory,     ///< The path was expected to direct to a directory, but doesn't.
+        IsADirectory,      ///< The path was expected to direct to non-directory link, but does.
+        NotEmpty,          ///< The directory specified is not empty.
+        // Other Errors
+        IOError,           ///< There was a hardware I/O error while completing the operation.
+        NoSpace,           ///< There is insufficient space on the filesystem to complete the operation.
+        MaxLinksExceeded,  ///< The operation would exceeded the maximum links supported for the directory.
+        CurrentlyBusy,     ///< The file or directory specified is in use by the system and operation can't complete.
+        OperationCanceled, ///< The operation was canceled or aborted.
+        // Unknown
+        Unknown            ///< Error is unknown.
+    };
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Basic File Management
+
+    /// @brief Verifies the existence of a file.
+    /// @param FilePath The path and name of the file to check for.
+    /// @return Returns true if the file at the specified location exists, false otherwise.
+    [[nodiscard]]
+    Boole MEZZ_LIB FileExists(const StringView FilePath);
+
+    /// @brief Copies a file on disk to a new location.
+    /// @note This function makes no attempt to copy file permissions or attributes, only data.
+    /// @param OldFilePath The existing path to the file (including the filename) to be copied.
+    /// @param NewFilePath The path (including the filename) to where the file should be copied.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully copied, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CopyFile(const StringView OldFilePath, const StringView NewFilePath,
+                                   const Boole FailIfExists);
+    /// @brief Moves a file on disk from one location to another.
+    /// @remarks This function can be used to rename files.
+    /// @param OldFilePath The existing path to the file (including the filename) to be moved.
+    /// @param NewFilePath The path (including the filename) to where the file should be named.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully moved, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB MoveFile(const StringView OldFilePath, const StringView NewFilePath,
```

Comment:
Should FailIfExists default to true? This is the safer option and would make using this easier?

Instead of a boole it could instead accept an enum instance of FileOverwriteParameters or ShouldOverwrite or something similar. This could look like `MoveFile("foo.txt", "bar.txt", FileOverwrite::NoOverwrite)` or `MoveFile("foo.txt", "bar.txt", ShouldOverwrite::No)`.  This should reduce errors of understanding boolean logic.

Reply:
Added a FileOverwrite enum with "Allow" and "Deny" values.  These replaced the bool parameters.